File: intro_to_NA/file/Advection_exact1D.m
%% Input Parameters
c = 1;
t = 0;
x = linspace(-3,10, 200);
ax = subplot(1,1,1);
plot(ax, x, exp(-(x-c*t).^2)); % Initial plot
xlim(ax, [-3,10]);ylim(ax, [-0.1,1.1]);
name = title(['t = ',num2str(t),'[s]']);
hold(ax ,"on")


for t = 0:1:15 % time data
    name = title(['t = ',num2str(t,'%04.1f'),'[s]']);
    plot(ax, x, exp(-(x-c*t).^2));
    hold(ax ,"on")
    % filename = sprintf('plotEXACT%04.1f.png', t); % file name 
    %saveas(fig, filename); % save figure as png
end

File: intro_to_NA/file/Advection_exact1D.py
# -*- coding: utf-8 -*-
import numpy as np
import matplotlib.pyplot as plt

## Input Parameters
c = 1
t_init = 0
t_max = 15
dt    = 0.2
x = np.linspace(-3, 10, 200, endpoint=True)
t = np.arange(t_init, t_max+dt, dt, )

## Plot Data
fig = plt.figure()
ax = fig.add_subplot(1,1,1)

for i in t:
  ax.plot(x, np.exp(-(x-c*i)**2), color = '#4169e1')
  ax.set_ylim(-0.1,1.1)
  ax.set_xlim(-3, 10)
  ax.set_title('t=%05.1f'%(i)+'[s]')
  # fig.savefig('Advection_exact%05.1f'%(i)+'.png') # Save figure
  # ax.cla()

plt.show()